address-space-controller/src/main/java/io/enmasse/controller/ControllerChain.java

```diff
@@ -102,7 +102,12 @@ public void onUpdate(List<AddressSpace> resources) throws Exception {
                 for (Controller controller : chain) {
                     log.info("Controller {}", controller);
                     log.debug("Address space input: {}", addressSpace);
-                    addressSpace = controller.reconcileAnyState(addressSpace);
+                    Controller.ReconcileResult result = controller.reconcileAnyState(addressSpace);
+                    addressSpace = result.getAddressSpace();
+                    // If instructed to requeue, break loop and replace if modified
+                    if (result.isPersistAndRequeue()) {
+                        break;
```

Comment:
Shouldn't there be some code which actually re-triggers the processing? I guess it would still work if you did make a change to the address space. But if you did not, then it would also not re-run. Which might be ok in this specific case.

Reply:
It is assumed, that if this happens, and you actually made a change to the object, then you call replace and will get a notification of the change, so I'm not sure if we need to explicitly re-trigger.

Reply:
I would just add the requirement to some javadoc :)